etc/testing/travis_test_check_and_pull.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+set -ex
+
+# See travis_build_check_and_stash.sh for context.
+#
+# This script takes the docker image which that script pushed, and unpacks the
+# git source tree from it, as it is more reliably going to be the correct
+# source tree for this build than what Travis and GitHub give us (in the PR
+# build case, at least).
+
+if [ "${TRAVIS_PULL_REQUEST}" == "false" ]; then
+    # These shenannigans not needed for release and branch builds, hopefully.
+    exit 0
+fi
+
+if [[ ! "$TRAVIS_SECURE_ENV_VARS" == "true" ]]; then
+    echo "Need travis env vars so we can auth to docker hub."
+    exit 1
+fi
```

Comment:
Note that PRs from outside contributors will not have access to secure env vars. Will this break their PRs?

Reply:
Good catch - I'll test this and fix any breakages

Reply:
Is this not already broken if build stage needs docker hub access?

Reply:
(and if that's the case, let's move it to a followup issue but definitely fix it soon)

Reply:
In the case of outside PRs, we don't push images, we just quickly exit the build step and run the build process in each test bucket

Reply:
got it - https://github.com/pachyderm/pachyderm/pull/5137/commits/9bf5f669cad5cd79735a2bae9e9f5a8ac23168e8 should fix this